src/block_coordinate_algorithms.jl

```diff
@@ -0,0 +1,361 @@
+abstract type BlockCoordinateMethod end
+
+function perform_bc_updates end
+
+abstract type UpdateOrder end
```

Comment:
More self-contained as a name: `BlockCoordinateUpdateOrder`  or something? So that someone who sees the type somewhere knows what the context is?